src/traits/LazyDataLoadTrait.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace luya\admin\traits;
+
+trait LazyDataLoadTrait
+{
+    /**
+     * @param array|\Closure $data
+     * @return array|\Traversable
+     */
+    protected function loadData($data)
+    {
+        if (is_callable($data)) {
+            return call_user_func($data);
+        } elseif (is_array($data) || $data instanceof \Traversable) {
+            return $data;
+        }
+        
+        return [];
```

Comment:
Is that correct? Is it only possible to handle array data? What about `User::find()->select(['name'])->scalar()`. So just return the input data. maybe only check for is_callable and return the input value otherwise?